redsky/streamer.py

```diff
@@ -0,0 +1,57 @@
+from uuid import uuid4
+import os
+from copy import deepcopy as dc
+
+
+def db_store(db, fs_data_name_save_map=None):
+    if fs_data_name_save_map is None:
+        fs_data_name_save_map = {}
+
+    def wrap(f):
+        def wrapped_f(*args, **kwargs):
+            gen = f(*args, **kwargs)
+            for name, doc in gen:
+                fs_doc = dc(doc)
+
+                if name == 'descriptor':
+                    # Mutate the doc here to handle filestore
+                    for data_name in fs_data_name_save_map.keys():
+                        fs_doc['data_keys'][data_name].update(
+                            external='FILESTORE:',
+                            dtype='array')
+
+                elif name == 'event':
+                    # Mutate the doc here to handle filestore
+                    for data_name, sub_dict in fs_data_name_save_map.items():
```

Comment:
Yes, sorry I should have fixed this. I think we should still have the folder in the `data_name_save_map` since I would like the ability to change the folder a bit more dynamically, rather than just at the save function define time.